src/Model/batchprocessing/BatchProcessFMAID2ROIName.py

```diff
@@ -0,0 +1,132 @@
+import csv
+from pydicom import dcmread
+from src.Controller.PathHandler import data_path
+from src.Model import ROI
+from src.Model.batchprocessing.BatchProcess import BatchProcess
+from src.Model.PatientDictContainer import PatientDictContainer
+
+
+class BatchProcessFMAID2ROIName(BatchProcess):
+    """
+    This class handles batch processing for the FMA ID to ROI name process.
+    Inherits from the BatchProcess class.
+    """
+    # Allowed classes for ROI Name Cleaning
+    allowed_classes = {
+        # RT Structure Set
+        "1.2.840.10008.5.1.4.1.1.481.3": {
+            "name": "rtss",
+            "sliceable": False
+        }
+    }
+
+    def __init__(self, progress_callback, interrupt_flag, patient_files):
+        """
+        Class initialiser function.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param patient_files: List of patient files.
+        """
+        # Call the parent class
+        super(BatchProcessFMAID2ROIName, self).__init__(progress_callback,
+                                                        interrupt_flag,
+                                                        patient_files)
+
+        # Set class variables
+        self.required_classes = ['rtss']
+        self.organ_names = {}
+        self.fma_ids = []
+        self.ready = self.load_images(patient_files, self.required_classes)
+        self.patient_dict_container = PatientDictContainer()
+
+    def start(self):
+        """
+        Goes through the steps of the ROI Name Cleaning process.
+        :return: True if successful, False if not.
+        """
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        # Lookup ROI names in Organ List
+        self.progress_callback.emit(("Reading ROIs...", 40))
+        roi_names = self.find_fma_ids()
+
+        # Return false if RTSS has no ROIs
+        if not roi_names:
+            self.summary = "ROI_NO_FMA"
+            return False
+
+        # Convert FMA ID to ROI name 
+        rtss = self.patient_dict_container.dataset['rtss']
+        total = 0
+        progress = 40
+        step = (90 - 40)/len(roi_names)
```

Comment:
Why have 90-40 instead of just 50?

Reply:
Fair comment, there is no reason. Changing this now.